packages/autorest.python/autorest/codegen/models/constant_type.py

```diff
@@ -77,6 +77,9 @@ def docstring_type(self, **kwargs: Any) -> str:
     def type_annotation(self, **kwargs: Any) -> str:
         return self.value_type.type_annotation(**kwargs)
 
+    def literal_annotation(self, **kwargs: Any) -> str:
```

Comment:
Instead of adding a `literal_annotation` to everything, should we always type constants as literal? Meaning we make the `def type_annotation` for constant types return a `Literal[{self.value_type}]`? I think it makes sense from what a constant represents

Reply:
yes, literal should be good for all constant except some special value types such as float.
A little complex to make the imports of Literal be clean on all cases, I will continue this PR tomorrow.

Reply:
why does literal not work with float?

Reply:
It's strange that mypy and intellisense don't support float Literal while the python execution does be OK.

The mypy complains for Literal[float]: 
![image](https://user-images.githubusercontent.com/59815250/195227148-18e9bc44-a23b-493b-a614-e38bfa87a802.png)
And the Intellisense also has below:
![image](https://user-images.githubusercontent.com/59815250/195227531-d88aafbd-c1a9-4372-9178-3d5205bb073f.png)
As saying in the mypy documentation: https://mypy.readthedocs.io/en/stable/literal_types.html#parameterizing-literals , the float is not available Literal.